src/google/protobuf/compiler/csharp/csharp_message.cc

```diff
@@ -98,12 +98,21 @@ const std::vector<const FieldDescriptor*>& MessageGenerator::fields_by_number()
   return fields_by_number_;
 }
 
+void MessageGenerator::AddDeprecatedFlag(io::Printer* printer) {
+  if (descriptor_->options().deprecated())
+  {
```

Comment:
One small thing: our usual C++ style is to put the opening curly brace on the same line as the `if (...)`.